### PR TITLE
fix(editor): Fix memory leak in Node Detail View by correctly unsubscribing from event buses

### PR DESCRIPTION
**Pull request by OlegIvaniv**
This PR addresses two issues related to eventBus subscriptions and the implementation of `onBeforeUnload` in NodeView.

1. Unsubscribing from eventBus: There's several instances where `eventBus.on` was used without properly unsubscribing using `eventBus.off`, or where `eventBus.off` was called without an appropriate handler. This PR ensures correct event bus unsubscription to prevent memory leaks.

2. Modifying `onBeforeUnload` in NodeView: The previous implementation involved appending the `onBeforeUnload` handler to the global window object, so it could later be disabled from e2e tests. Instead, we now check if the `preventNodeViewBeforeUnload` property exists in the window object, and terminate the handler if true.

Github issue / Community forum post (link here to close automatically):

### CODE REVIEW

File: packages/editor-ui/src/components/NodeSettings.vue [hunk 1]
```diff
@@ -895,18 +895,24 @@ export default mixins(externalHooks, nodeHelpers).extend({
...
 	},
 	mounted() {
 		this.populateHiddenIssuesSet();
 		this.setNodeValues();
 		if (this.eventBus) {
-			this.eventBus.on('openSettings', () => {
-				this.openPanel = 'settings';
-			});
+			this.eventBus.on('openSettings', this.openSettings);
 		}
```
Comment (netroy) [inline packages/editor-ui/src/components/NodeSettings.vue:907]:
```
this.eventBus?.on('openSettings', this.openSettings);
```

File: packages/editor-ui/src/components/NodeSettings.vue [hunk 2]
```diff
@@ -895,18 +895,24 @@ export default mixins(externalHooks, nodeHelpers).extend({
...
 
 		this.updateNodeParameterIssues(this.node as INodeUi, this.nodeType);
 	},
+	destroyed() {
+		if (this.eventBus) {
+			this.eventBus.off('openSettings', this.openSettings);
+		}
```
Comment (netroy) [inline packages/editor-ui/src/components/NodeSettings.vue:914]:
```suggestion
		this.eventBus?.off('openSettings', this.openSettings);
```

**Review by netroy — CHANGES_REQUESTED** [review]

**Review by netroy — DISMISSED** [review]
LGTM

**Review by netroy — DISMISSED** [review]

File: packages/editor-ui/src/views/SettingsLogStreamingView.vue [hunk 3]
```diff
@@ -173,6 +171,13 @@ export default mixins().extend({
 		},
 	},
 	methods: {
+		onDestinationWasSaved() {
+			this.$forceUpdate()
```
Comment (netroy) [inline packages/editor-ui/src/views/SettingsLogStreamingView.vue:175]: linting complaining about the missing semicolon here

Comment (OlegIvaniv) [reply packages/editor-ui/src/views/SettingsLogStreamingView.vue]: Damnit, my VSCode eslint keeps failing me 😠 Should be fixed now

File: cypress/e2e/14-data-transformation-expressions.cy.ts [hunk 4]
```diff
@@ -12,11 +12,12 @@ describe('Data transformation expressions', () => {
...
 		wf.actions.visit();
 
-		cy.window()
-			// @ts-ignore
-			.then(
-				(win) => win.onBeforeUnloadNodeView && win.removeEventListener('beforeunload', win.onBeforeUnloadNodeView),
-			);
+		cy.window().then(
+			(win) => {
+				// @ts-ignore
```
Comment (netroy) [inline cypress/e2e/14-data-transformation-expressions.cy.ts:17]: could we augment the typeof `win` to have `preventNodeViewBeforeUnload?: boolean`, so we don't need to sprinkle `// @ts-ignore` in so many places ?

**Review by netroy — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@0.224.2`](https://github.com/n8n-io/n8n/releases/tag/n8n%400.224.2)

**Pull request merged**